src/main/java/com/checkmarx/flow/service/ResultsService.java

```diff
@@ -171,43 +169,65 @@ void processResults(ScanRequest request, ScanResults results) throws MachinaExce
             log.info("######################################");
         }
     }
 
+    private void handleCustomIssueCase(ScanRequest request, ScanResults results) throws MachinaException {
+        try {
+            log.info("Issue tracking is custom bean implementation");
+            issueService.process(results, request);
+        } catch (HttpClientErrorException e) {
+            if (e.getRawStatusCode() == HttpStatus.UNAUTHORIZED.value()) {
+                throw new GitHubClientRunTimeException("GitHub token is invalid\n" + e.getMessage());
+            } else {
+                throw e;
+            }
+        }
+    }
+
     private void handleJiraCase(ScanRequest request, ScanResults results) throws JiraClientException {
         try {
-            log.info("Processing results with JIRA issue tracking");
+            log.info("======== Processing results with JIRA issue tracking ========");
             jiraService.process(results, request);
         } catch (RestClientException e) {
-            if (e.getStatusCode().isPresent() && e.getStatusCode().get() ==  HttpStatus.NOT_FOUND.value()) {
-                throw new JiraClientRunTimeException("Jira service is not accessible for URL: " + jiraService.getJiraURI(), e);
-            } else {
-                Map<String, ScanResults.XIssue> nonPublishedScanResultsMap = jiraService.getNonPublishedScanResults();
-                if (e.getStatusCode().isPresent() &&
-                        e.getStatusCode().get() ==  HttpStatus.BAD_REQUEST.value() &&
-                        nonPublishedScanResultsMap.size() > 0) {
-                    throwExceptionWhenPublishingErrorOccurred(e, nonPublishedScanResultsMap);
-
-                } else {
-                    throw  e;
-                }
-            }
+            handleJiraRestClientException(e);
         } catch (JiraClientException e) {
+            handleJiraClientException(e);
+        }
+    }
+
+    private void handleJiraClientException(JiraClientException e) throws JiraClientException {
+        Map<String, ScanResults.XIssue> nonPublishedScanResultsMap = jiraService.getNonPublishedScanResults();
+        if (nonPublishedScanResultsMap.size() > 0) {
+            throwExceptionWhenPublishingErrorOccurred(e, nonPublishedScanResultsMap);
+        } else {
+            throw e;
+        }
+    }
+
+    private void handleJiraRestClientException(RestClientException e) {
+        if (e.getStatusCode().isPresent() && e.getStatusCode().get() == HttpStatus.NOT_FOUND.value()) {
+            throw new JiraClientRunTimeException("Jira service is not accessible for URL: " + jiraService.getJiraURI() + "\n" + e.getMessage());
+        } else if (e.getStatusCode().isPresent() && e.getStatusCode().get() == HttpStatus.FORBIDDEN.value()) {
+            throw new JiraClientRunTimeException("Access is forbidden. Please check your basic auth Token \n" + e.getMessage());
```

Comment:
Instead of e.getMessage(), you can add 'e' as a second parameter to exception constructor.